src/walletClient/ClientMethods.js

```diff
@@ -249,6 +249,31 @@ exports.UserItems = async function({sortBy="default"}={}) {
   return this.FilteredQuery({mode: "owned", sortBy, ...(arguments[0] || {})});
 };
 
+/**
+ * <b><i>Requires login</i></b>
+ *
+ * Retrieve all NFTs owned by the current user, with their full info.
+ *
+ * @methodGroup User
+ * @namedParams
+ *
+ * @returns {Promise<Object>} - Results of the query and pagination info
+ */
+exports.UserNFTs = async function() {
```

Comment:
I don't particularly like the name of this because we already have 'UserItems' and it's confusing to have two different apis that appear to do the same thing

what exactly is the difference between this API and the user items api? (as/wlt/:addr)

maybe we can modify UserItems to take a parameter to use this api instead

Reply:
It's kind of a combo of calling `UserItems`, and the `NFT` for every item

Reply:
does it happen to support the same / similar filter/sort/page parameters to the user items nft? (if not it really ought to)

I would like to make it so that we can just add a parameter 'includeFullMetadata' that switches to this API instead of the other one. 

we could potentially also switch to this API entirely, but the increased data size might not be worth it and I'm not sure how it retrieves the metadata exactly and that's important. basically, nft metadata lives in two places: /public/nft (token metadata) and /public/asset_metadata/nft (full metadata). for the most part, token metadata is just a subset of the full metadata, but for certain cases (e.g. generative nfts where each token metadata is different) it's important that the token metadata is merged over the full metadata

anyway, let's just make this an option of UserItems instead of a new method. might need some improvements to the API to match the options available for the other API, but we can get to that.

Reply:
@elv-todd, does it support the same filter/sort?

Reply:
yep it's an extended version of /as/wlt/:addr, accepts the same sort and filter args.

Reply:
doesn't -- just adds a raw "nft_template" section in each contents item.  don't think we realized that needed to be overridden from the token_uri based on wayne's iOS sample..

for the overrides, can i just put all the toplevel keys into the toplevel of the template, or, does it need to deal with merging subtrees?

Reply:
deep merge

the other API returns metadata from the token url which, for generative, pulls from /public/nft/:tokenId. we need that to override values from public/asset_metadata/nft, otherwise they won't show up correctly. it's possible we only need to do this for generative NFTs, but I'm not positive. the UI (client) always does this merge for individual NFTs, so we probably should do it for the listing views as well.

here's a user with a bunch of generative walking dead nfts for reference:
https://wallet.contentfabric.io/#/marketplace/iq__2ApMZ9hdFmbL314wrba1ZxfSgRKL/users/0xc77c1fb54c767d343942aa538fc6fbc678519187/listings

Reply:
ok confirmed...   using that Maggie Walking Dead one, 

https://main.net955305.contentfabric.io/s/main/q/hq__BbSAc2xMorYJsMm84NP9jkU1oik3Be5oVGB91zsAJL4SParPp9TEiuHAisdLAWTmpLzqyPZ1Bp/meta/public/asset_metadata/nft

needs to override with

https://main.net955305.contentfabric.io/s/main/q/hq__BbSAc2xMorYJsMm84NP9jkU1oik3Be5oVGB91zsAJL4SParPp9TEiuHAisdLAWTmpLzqyPZ1Bp/meta/public/nfts/5253

to test -- asset_metadata json has generic "image": https://main.net955305.contentfabric.io/s/main/q/hq__3fB8tCPsQmNyej1T8dDmB7dn5ic2aezbXXFAJ2HmtXGoDxgmNuD8jL7f4pH9eCFiN1ppmmxqeS/files/TWD-SURVIVORS-SERIES-PREVIEW--STATIC--V1-q20.jpg

numbered nfts/:tokenID has maggie "image": https://main.net955305.contentfabric.io/s/main/q/hq__Cvvo9vyS1Lmfj9tv5tpRnvVbqm427qaiZQgjAYjix72yAEqUfSF3Hou3mUNA5KPuab5UUrniDY/files/nft/MA_NO_147.jpg

and then we look at the output of the wallet query to make sure we get the overridden "image":; and we do.

so looks like the blockchain indexer is handling this for us properly, and we inherit it in the as/wlt/:addr which then means apigw/nfts inherits it too.